pkg/clusters/types/eks/aws-operations/eks.go

```diff
@@ -0,0 +1,479 @@
+package aws_operations
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/pkg/errors"
+	"github.com/samber/lo"
+	"github.com/weaveworks/eksctl/pkg/ami"
+	eksctlapi "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/authconfigmap"
+	eksiam "github.com/weaveworks/eksctl/pkg/iam"
+	"github.com/weaveworks/eksctl/pkg/nodebootstrap"
+	"k8s.io/client-go/kubernetes"
+	"os"
+	"time"
+)
+
+const (
+	DefaultNodeGroupName     = "default-node-group"
+	DefaultKubernetesSvcCIDR = "172.20.0.0/16"
+	kubernetesTagFormat      = "kubernetes.io/cluster/%s"
+	envKeyNodeSSHKeyName     = "EKS_NODE_SSH_KEY"
+
+	TagNameCreateBy = "ktf_created_by"
+)
+
+func CreateEKSClusterAll(ctx context.Context, cfg aws.Config, clusterName,
+	k8sMinorVersion, nodeMachineType string, tags map[string]string) error {
+
+	stsClient := sts.NewFromConfig(cfg)
+	ec2Client := ec2.NewFromConfig(cfg)
+	eksClient := eks.NewFromConfig(cfg)
+	iamClient := iam.NewFromConfig(cfg)
+
+	callIdentityOutput, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get caller identity")
+	}
+	createdByArn := aws.ToString(callIdentityOutput.Arn)
+
+	clusterRoleArn, nodeRoleArn, err := createRoles(ctx, iamClient, clusterName)
+	if err != nil {
+		return errors.Wrap(err, "failed to create IAM roles")
+	}
+	subnetAvZones, err := getAvailabilityZones(ctx, ec2Client, cfg.Region)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get availability zones in region %s", cfg.Region)
+	}
+
+	vpcId, subnetIDs, err := createVPC(ctx, ec2Client, subnetAvZones)
```

Comment:
Could it leave dangling resources when error happens after creating a VPC? I think we need to clean them if possible, or add a comment to tell that it may leave resource to clear.

Reply:
Yes, there can be these orphan resources left when the cluster creation is not complete. I'll provide a better way of cleaning up by enumerate tags in a future PR.

Reply:
The creation can fail because of different reasons:
* the user does not have permission to perform some of the actions
* API call fails because of network issue

So we'll also need to enhance the experence of these factors.

Reply:
I'll mark this as  resolved for now as it's not planned in this PR.